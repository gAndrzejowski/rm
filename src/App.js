// @flow
import React, { Fragment } from 'react';
import { Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import Heading from './components/Heading/Heading';
import Footing from './components/Footing/Footing';
import NoMovies from './components/NoMovies/NoMovies';
import ErrorBoundary from './components/ErrorBoundary/ErrorBoundary';
import NotFound from './components/NotFound/NotFound';
import Related from './components/Related/Related';
import SearchResults from './components/SearchResults/SearchResults';
import styles from './App.scss';

type Props = {
  Router :(any) => any,
  location?: string,
  context?: Object,
  Store: Object
};

const App = ({
  Router, location, context, Store,
} :Props) => (
  <Provider store={Store}>
    <Router location={location} context={context}>
      <Fragment>
        <ErrorBoundary>
          <Heading />
        </ErrorBoundary>
        <ErrorBoundary>
          <Switch>
            <Route exact path="/" component={NoMovies} />
            <Route exact path="/film/:id" component={Related} />
            <Route exact path="/search/:by/:query" component={SearchResults} />
            <Route path="*" component={NotFound} />
          </Switch>
        </ErrorBoundary>
        <ErrorBoundary>
          <Footing />
        </ErrorBoundary>
      </Fragment>
    </Router>
  </Provider>
);

export default App;
