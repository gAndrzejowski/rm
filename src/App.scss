.subscript {
  color:navy;
  font-size:14pt;
}
body, html {
  margin:0 auto;
  background-color:#222;
}