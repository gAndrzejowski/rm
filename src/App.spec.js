import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import App from './App';

jest.mock('react-dom');

describe('App component', () => {

    it('renders correctly', () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('initializes state', () => {
        const wrapper = shallow(<App />);
        expect(wrapper.state('selectedMovie')).toBeNull();
        expect(wrapper.state('currentSort')).toBe('release_date');
        expect(wrapper.state('results')).toEqual(expect.any(Array));
    });
    it('has chooseMovie and resetMovie methods that handle selectedMovie state', () =>{
        const wrapper = shallow(<App />);
        wrapper.instance().chooseMovie('X');
        expect(wrapper.state('selectedMovie')).toBe('X');
        wrapper.instance().resetMovie();
        expect(wrapper.state('selectedMovie')).toBeNull();
    });
    it('has changeSorting method that handles currentSort state', () => {
        const wrapper = shallow(<App />);
        wrapper.instance().changeSorting('some_criterion');
        expect(wrapper.state('currentSort')).toBe('some_criterion');
    });
    it('has a getSortingFunc method that returns function to sort an array of objects by currentSort property in descending order', () => {
        const wrapper = shallow(<App />);
        const toSort = [{a:2, b:3}, {a:3, b:1}, {a:1, b:2}];
        wrapper.instance().changeSorting('a');
        let sorter = wrapper.instance().getSortingFunc();
        expect(toSort.sort(sorter)).toEqual([{a:3, b:1}, {a:2, b:3}, {a:1, b:2}]);
        wrapper.instance().changeSorting('b');
        sorter = wrapper.instance().getSortingFunc();
        expect(toSort.sort(sorter)).toEqual([{a:2, b:3}, {a:1, b:2}, {a:3, b:1}]);
    });
});