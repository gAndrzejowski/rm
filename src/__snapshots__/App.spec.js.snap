// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<Provider
  store={undefined}
>
  <UNDEFINED
    context={undefined}
    location={undefined}
  >
    <UNDEFINED>
      <ErrorBoundary>
        <Heading />
      </ErrorBoundary>
      <ErrorBoundary>
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/film/:id"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/search/:by/:query"
          />
          <Route
            component={[Function]}
            path="*"
          />
        </Switch>
      </ErrorBoundary>
      <ErrorBoundary>
        <Footing />
      </ErrorBoundary>
    </UNDEFINED>
  </UNDEFINED>
</Provider>
`;
